psp/exp_configs/island_exc.py

```diff
@@ -0,0 +1,109 @@
+from psp.data_sources.pv import NetcdfPvDataSource, PvDataSource
+from psp.dataset import PvSplits, auto_date_split, split_pvs
+from psp.exp_configs.base import ExpConfigBase
+from psp.models.base import PvSiteModel, PvSiteModelConfig
+from psp.models.recent_history import RecentHistoryModel
+from psp.models.regressors.decision_trees import SklearnRegressor
+from psp.typings import Horizons
+
+PV_TARGET_DATA_PATH = "/mnt/storage_b/data/ocf/solar_pv_nowcasting/clients/island/pv_hourly_v6.nc"
+GFS_NWP_PATH = "/mnt/storage_b/data/ocf/solar_pv_nowcasting/clients/island/5y_nwp_3_to_48.zarr"
+EXC_PATH = (
+    "/mnt/storage_b/data/ocf/solar_pv_nowcasting/experimental/Excarta/"
+    "sr_UK_Malta_full/zarr_format/r4/Malta/merged_full/malta_excarta_hour_shift.zarr"
+)
+
+
+def _get_capacity(data: xr.Dataset) -> float:
+    # Use the "capacity" data variable.
+    # There is one capacity per timestamp, let's take the first one arbitrarily; hopefully there
+    # is not a big variation of capacity in our history of a few days!
+    return float(data["capacity"].values[0])
+
+
+class ExpConfig(ExpConfigBase):
+    def get_pv_data_source(self):
+        return NetcdfPvDataSource(
+            PV_TARGET_DATA_PATH,
+        )
+
+    def get_data_source_kwargs(self):
+        return dict(
+            pv_data_source=NetcdfPvDataSource(
+                PV_TARGET_DATA_PATH,
+                lag_minutes=5 * 24 * 60,
+            ),
+            nwp_data_source=[
+                NwpDataSource(
+                    GFS_NWP_PATH,
+                    coord_system=4326,
+                    x_dim_name="latitude",
+                    y_dim_name="longitude",
+                    x_is_ascending=False,
+                ),
+                NwpDataSource(
+                    EXC_PATH,
+                    coord_system=4326,
+                    x_dim_name="x",
+                    y_dim_name="y",
+                    time_dim_name="init_time",
+                    x_is_ascending=True,
+                    single_point=True,
+                    lag_minutes=8 * 60,
+                ),
+            ],
+        )
+
+    def get_model_config(self):
+        return PvSiteModelConfig(
+            horizons=Horizons(
+                duration=60,
+                num_horizons=48,
+            )
+        )
+
+    def get_model(self, **kwargs) -> PvSiteModel:
+        return RecentHistoryModel(
+            self.get_model_config(),
+            **self.get_data_source_kwargs(),
+            regressor=SklearnRegressor(
+                num_train_samples=10000,
+                normalize_targets=True,
+            ),
+            # EXAMPLE USE for use_nwp:
+            # use_nwp = None: don't use any NWP sources
+            # use_nwp = ["gfs", None]: use just GFS
+            # use_nwp = [None, "excarta"]: use just Excarta
+            # use_nwp = ["gfs", "excarta"]: use both
+            # It is important to retain the order of NWPs that was set above
```

Comment:
This is a bad idea. Too many ways to shoot yourself in the foot and very confusing. There are many good options I think:
1. Use lists like you are doing but use two variables for names and "use_nwp" (`nwp_names: list[str]`, `use_nwp: list[bool]`).
2. Use dictionaries instead of lists for nwps: `nwp_data_sources: dict[str, NwpDataSource]`, `nwp_variables: dict[str, list[str]]`, etc.


My recommendation:
* Use # 2 if you care about naming the nwps
* Drop `use_nwp` altogether. As @jacobbieker pointed out it's not really useful, I'm not sure why I put it here in the first place.
* Only use scalars for `nwp_dropout` and `nwp_tolerance`. It will simplify the code and I don't see a good use-cases where you need to specify per-nwp values for those. If that happens later it will be easy to add.

Reply:
Thanks for the feedback Simon. I have dropped how the use_nwp is currently used and moved it to be an additional variable in the NwpDataSource class instead. A dictionary seems the way to go as i want to ensure the labelling of nwps, especially as we start to introduce more variables.